commown_devices/tests/test_project_task.py

```diff
@@ -425,3 +463,40 @@ def test_outward_inward_notracking(self):
             final_location,
             self.env.ref("commown_devices.stock_location_devices_to_check"),
         )
+
+    def test_change_task_check(self):
+        with self.assertRaises(ValidationError) as err:
+            self.task_test_checks.stage_id = self.ongoing_stage
+        self.assertEqual(
+            "Error while validating constraint\n\nThis task can not be moved forward. There are no picking linked to this task.\n",
+            err.exception.name,
+        )
+        with self.assertRaises(ValidationError) as err2:
+            self.task_test_checks2.stage_id = self.ongoing_stage
+        self.assertEqual(
+            "Error while validating constraint\n\nThis task can not be moved forward. There are no picking linked to this task.\n",
+            err2.exception.name,
+        )
+
+        module = self.nontracked_product.product_variant_id
+        lot = self.storable_product.product_variant_id
+
+        quant = self.env["stock.quant"].search(
+            [
+                ("product_id", "=", lot.id),
+                ("quantity", ">", 0),
+                ("location_id.complete_name", "ilike", "WH/STOCK/"),
```

Comment:
wahou, ça marche ça ? Il est stored le champs complete_name ? Cool. J'aurais utilisé child_of, c'est un poil mieux que de se baser sur le nom des emplacements (qui pourrait changer).

Reply:
Alors oui ça marche mais +1 pour le child of c'est plus propre !